master/buildbot/util/git.py

```diff
@@ -137,66 +159,112 @@ def ensureSshKeyNewline(privateKey: str) -> str:
 
 
 class GitStepMixin(GitMixin):
+    _git_auth: GitStepAuth
+
     def setupGitStep(self):
-        self.didDownloadSshPrivateKey = False
-        self.setupGit(logname='Git')
+        self.setupGit()
 
         if not self.repourl:
             config.error("Git: must provide repourl.")
 
-    def _isSshPrivateKeyNeededForGitCommand(self, command):
-        if not command or self.sshPrivateKey is None:
-            return False
+        if not hasattr(self, '_git_auth'):
+            self._git_auth = GitStepAuth(self)
+
+    def setup_git_auth(
+        self,
+        ssh_private_key: IRenderable | None,
+        ssh_host_key: IRenderable | None,
+        ssh_known_hosts: IRenderable | None,
+    ) -> None:
+        self._git_auth = GitStepAuth(
+            self,
+            ssh_private_key,
+            ssh_host_key,
+            ssh_known_hosts,
+        )
 
-        gitCommandsThatNeedSshKey = ['clone', 'submodule', 'fetch', 'push']
-        if command[0] in gitCommandsThatNeedSshKey:
-            return True
-        return False
+    @property
+    def sshPrivateKey(self):
+        ssh_auth_warndeprecate(
+            "self.sshPrivateKey",
+            "self._git_auth.ssh_private_key",
+        )
+        return self._git_auth.ssh_private_key
 
-    def _getSshDataPath(self):
-        # we can't use the workdir for temporary ssh-related files, because
-        # it's needed when cloning repositories and git does not like the
-        # destination directory being non-empty. We have to use separate
-        # temporary directory for that data to ensure the confidentiality of it.
-        # So instead of
-        # '{path}/{to}/{workerbuilddir}/{workdir}/.buildbot-ssh-key'
-        # we put the key in
-        # '{path}/{to}/.{workerbuilddir}.{workdir}.buildbot/ssh-key'.
+    @property
+    def sshHostKey(self):
+        ssh_auth_warndeprecate(
+            "self.sshHostKey",
+            "self._git_auth.ssh_host_key",
+        )
+        return self._git_auth.ssh_host_key
 
-        # basename and dirname interpret the last element being empty for paths
-        # ending with a slash
-        path_module = self.build.path_module
+    @property
+    def sshKnownHosts(self):
+        ssh_auth_warndeprecate(
+            "self.sshKnownHosts",
+            "self._git_auth.ssh_known_hosts",
+        )
+        return self._git_auth.ssh_known_hosts
 
-        workerbuilddir = bytes2unicode(self.build.builder.config.workerbuilddir)
-        workdir = self._getSshDataWorkDir().rstrip('/\\')
+    @abc.abstractmethod
+    def _get_auth_data_workdir(self) -> str:
+        raise NotImplementedError()
 
-        if path_module.isabs(workdir):
-            parent_path = path_module.dirname(workdir)
-        else:
-            parent_path = path_module.join(self.worker.worker_basedir, path_module.dirname(workdir))
+    def _getSshDataWorkDir(self) -> str:
```

Comment:
We're moving code around, so prefer changing the function names to `snake_case`.

Reply:
Ah, actually this also can be removed.